### PR TITLE
visualvm: 1.3.9 -> 1.4

### PR DESCRIPTION
**Pull request by moaxcp**
###### Motivation for this change
update version

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/java/visualvm/default.nix [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
+{ stdenv, fetchzip, lib, makeWrapper, makeDesktopItem, jdk, gtk2, gawk }:
 
 stdenv.mkDerivation rec {
-  name = "visualvm-1.3.9";
+  name = "visualvm-1.4";
 
   src = fetchzip {
-    url = "https://github.com/visualvm/visualvm.src/releases/download/1.3.9/visualvm_139.zip";
-    sha256 = "1gkdkxssh51jczhgv680i42jjrlia1vbpcqhxvf45xcq9xj95bm5";
+    url = "https://github.com/visualvm/visualvm.src/releases/download/1.4/visualvm_14.zip";
```
Comment (joachifm) [inline pkgs/development/tools/java/visualvm/default.nix]: Not a blocker but you could attempt to re-use the `version` attribute here to simplify bumps. It seems you could simply remove the dots in the version string for the archive basename.

**Comment by joachifm** [comment]
@GrahamcOfBorg build visualvm

File: pkgs/development/tools/java/visualvm/default.nix (hunk 1 above)
Comment (moaxcp) [reply pkgs/development/tools/java/visualvm/default.nix]: Ok I will try it out.

Comment (moaxcp) [reply pkgs/development/tools/java/visualvm/default.nix]: You know I just realized that I am only changing the url. How do I remove the sources from the nix store to ensure the url actually works when I run nix-build?

File: pkgs/development/tools/java/visualvm/default.nix [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
 
   src = fetchzip {
-    url = "https://github.com/visualvm/visualvm.src/releases/download/1.3.9/visualvm_139.zip";
-    sha256 = "1gkdkxssh51jczhgv680i42jjrlia1vbpcqhxvf45xcq9xj95bm5";
+    url = "https://github.com/visualvm/visualvm.src/releases/download/${version}/visualvm_${builtins.replaceStrings ["."] [""]  version}.zip";
+    sha256 = "0qpkal3yc8c8di4dmdglq54a4m7hciqximn5n6hwx2zqnfwx909z";
+  };
+
+  desktopItem = makeDesktopItem {
+      name = "visualvm";
```
Comment (moaxcp) [inline pkgs/development/tools/java/visualvm/default.nix]: Any ideas on how to add the icon? There is an icon located at etc/visualvm.icns. Is it ok to just set icon to that location?

Comment (moaxcp) [reply pkgs/development/tools/java/visualvm/default.nix]: I'm just wondering if an icns file is ok or does it need to be a png?

File: pkgs/development/tools/java/visualvm/default.nix (hunk 1 above)
Comment (joachifm) [reply pkgs/development/tools/java/visualvm/default.nix]: The quick way is to just perturb the hash while you're trying things out ... or maybe use `nix-prefetch-url -A ...` which should ensure the fetcher actually runs. Anyway, this is not all that important, so I'd be happy to integrate this as-is,. the version string reuse can be iterated upon later.

File: pkgs/development/tools/java/visualvm/default.nix (hunk 2 above)
Comment (joachifm) [reply pkgs/development/tools/java/visualvm/default.nix]: Per https://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html the supported formats are png, xpm, and svg.

Comment (moaxcp) [reply pkgs/development/tools/java/visualvm/default.nix]: I can't find a png file in the source. Would it be bad to convert the image to png using image magick?

**Comment by joachifm** [comment]
@GrahamcOfBorg build visualvm

File: pkgs/development/tools/java/visualvm/default.nix (hunk 2 above)
Comment (joachifm) [reply pkgs/development/tools/java/visualvm/default.nix]: No idea, sorry

**Comment by joachifm** [comment]
Thank you

**Pull request merged**